Library/Backends/ZendEngine2/Backend.php

```diff
@@ -253,17 +264,39 @@ public function onPostCompile($method, CompilationContext $context)
         $codePrinter = $context->codePrinter;
         if (preg_match('/^zephir_init_properties/', $method->getName())) {
             $codePrinter->increaseLevel();
-            $codePrinter->output('return Z_OBJVAL_P(this_ptr);');
+            $codePrinter->output('return retval;');
             $codePrinter->decreaseLevel();
             $codePrinter->output('}');
             $codePrinter->decreaseLevel();
         }
+        else {
+            $cleanup = '';
+            foreach ($context->symbolTable->getVariables() as $variable) {
+                if ($variable->getName() != 'this_ptr' && $variable->getName() != 'return_value' && $variable->getName() != 'return_value_ptr') {
+                    if ($variable->isComplex() && $variable->mustInitNull() && $variable->isLocalOnly()) {
+                        $cleanup .= "\t" . 'zval_dtor(&' . $variable->getName() . ');' . PHP_EOL;
```

Comment:
Could we incapsulate this into something like 
```php
$variable->haveToBeCleaned();
```

I'd like to cover this by tests and have an ability to mock this